cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -392,6 +510,17 @@ public Service generateHeadlessService() {
      * @return The generate StatefulSet
      */
     public StatefulSet generateStatefulSet(boolean isOpenShift) {
+        // TODO: Delete this?
+        return generateStatefulSet(isOpenShift, Collections.EMPTY_MAP);
+    }
+
+    /**
+     * Generates a StatefulSet according to configured defaults
+     * @param isOpenShift True iff this operator is operating within OpenShift.
+     * @return The generate StatefulSet
+     */
+    public StatefulSet generateStatefulSet(boolean isOpenShift, Map<String, String> externalAddresses) {
+        setExternalAddresses(externalAddresses);
```

Comment:
Well ... We need to pass this to the `getEnvVars` method. But it cannot be done through the method calls because they inherit from `AbstractModel`. So I use class variable to passed that way. If you think it cleaner, I can call `setExternalAddress` from `KafkaAssemblyOperator` directly if you think its cleaner.

Reply:
I think it's OK to leave it for now, but it points again to the inadequacy of `AbstractModel`.